packages/python/readme_metrics/PayloadBuilder.py

```diff
@@ -57,15 +57,19 @@ def __call__(self, request: Request, response: ResponseInfoWrapper) -> dict:
         Returns:
             dict: Payload object (ready to be serialized and sent to ReadMe)
         """
+        group = self.grouping_function(request)
+        if "api_key" in group:
+            group["id"] = group["api_key"]
+            del group["api_key"]
         payload = {
-            "group": self.grouping_function(request),
+            "group": group,
             "clientIPAddress": request.remote_addr,
             "development": self.development_mode,
             "request": {
                 "log": {
                     "creator": {
                         "name": __name__,
-                        "version": pkg_resources.require("readme_metrics")[0].version,
+                        "version": importlib.import_module(__package__).__version__,
```

Comment:
Why the switch to this?

Reply:
This wasn't working but because we were skipping tests we never noticed it. Once we started running tests, specifically for PayloadBuilder, then it because apparent. This appears to fix this issue!